edgelet/edgelet-hsm/src/crypto.rs

```diff
@@ -183,4 +183,11 @@ impl CoreCertificate for Certificate {
             .map_err(|err| Error::from(err.context(ErrorKind::Hsm)))
             .map_err(|err| CoreError::from(err.context(CoreErrorKind::KeyStore)))
     }
+
+    fn get_common_name(&self) -> Result<String, CoreError> {
+        self.0
+            .get_common_name()
+            .map_err(|err| Error::from(err.context(ErrorKind::Hsm)))
+            .map_err(|err| CoreError::from(err.context(CoreErrorKind::KeyStore)))
```

Comment:
I see that you're following the pattern from the other existing methods, but is somewhat curious that we are using 2 `map_err` calls to do this transformation. Would having only the second `map_err` be acceptable? Just that you wouldn't have the `ErrorKind::Hsm` error in the error stack.

Reply:
I think I prefer the HSM in the stack to quickly narrow down the scope of the issue. I did however amend the ErrorKind to be a bit more accurate.